lib/libvfio-user.c

```diff
@@ -1838,6 +1844,10 @@ vfu_map_sg(vfu_ctx_t *vfu_ctx, dma_sg_t *sg, struct iovec *iov, int cnt,
         return ERROR_INT(EINVAL);
     }
 
+    if (unlikely(vfu_ctx->pending == VFU_CTX_PENDING_DMA_UNMAP)) {
```

Comment:
It's a simplification, otherwise dirty page tracking gets complicated. Currenlty, we generate the dirty pages bitmap before we call the callback, so if the user calls `vfu_sg_map` we'll miss dirty pages. Do you think it's worth fixing now?

Reply:
I think we can revisit in context of #594 and #598

Reply:
and maybe we should reconsider #595 there too